### PR TITLE
ifconfig and ip reference

### LINKED ISSUE #299: Creating reference page for [ifconfig, ip]
Didn't see too many networking commands so requesting to add ifconfig.

### PR DESCRIPTION
**Pull request by zhecht**
# Advanced HW 5 submission

Closes #299 

Name: Zack Hecht
Uniqname: zhecht

### CODE REVIEW

File: _commands/tools/ip.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+---
+---
+ip
+----
+
+`ip` is a powerful networking tool that when simply used will display information on kernel-resident interfaces. Much like `ifconfig`, it can turn devices on/off and set IP and MAC addresses. `ip` also wraps popular commands like netstat, ndp, arp, and route into its functionality.
```
Comment (mterwill) [inline _commands/tools/ip.md]: Could you be a little more consistent with what's `code` and what's not? e.g. the commands in the last sentence.

Comment (mterwill) [inline _commands/tools/ip.md]: Could you be a little more consistent with what's code and what's not? e.g. the commands in the last sentence.

File: _commands/tools/ifconfig.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+---
+---
+
+ifconfig
+--
+
+`ifconfig` is used to configure a network interface or display information about all active or inactive connections.
+
+<div class="alert alert-warning">
+Warning: While ifconfig is still used, it is recommended to use iproute2 (Linux) or iproute2mac <i style="font-weight:bold;">[See reference on 'ip'].</i> ip is a much more powerful tool that can do everything ifconfig can do, and lots more.
```
Comment (mterwill) [inline _commands/tools/ifconfig.md]: Why don't you link directly to the `ip` page here?

File: _commands/tools/ifconfig.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Without an argument, `ifconfig` will print all active interfaces.
+
+`ifconfig -a` will print all inactive interfaces along with the active ones.
+
+An interface is usually a driver type followed by its unit number. 
+
+For example `eth0` resolves to the first ethernet interface and `en0` to the first wireless one.
+
+##### `*On Linux, wireless interfaces are called wlan0,wlan1,etc.*`
```
Comment (mterwill) [inline _commands/tools/ifconfig.md]: This isn't code.

File: _commands/tools/ifconfig.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  ether AA:BB:CC:DD:EE:FF
+  inet6 fe80::4cd:32c1:df16:a80a%en0 prefixlen 64 secured scopeid 0x4 
+  inet 192.168.1.8 netmask 0xffffff00 broadcast 192.168.1.25
+  nd6 options=201<PERFORMNUD,DAD>
+  media: autoselect
+  status: active
+
+#The machine has a wireless connection that is active, up, and running, with an IP address of 192.168.1.8.
+#ether refers to the MAC address of the machine.
+#mtu refers to the Maximum Transmission Unit or the limit on the size of packets sent.
```
Comment (mterwill) [inline _commands/tools/ifconfig.md]: Maybe take these explanations out of the code block?

File: _commands/tools/ip.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+`iproute2mac` is available through Homebrew as a compatible version. 
+
+It is NOT fully compatible though. It's source and documentation can be found on github <a href="https://github.com/brona/iproute2mac" target="_blank">here.</a>
+
+Also, `en0` is the equivalent to Linux's `wlan0` wireless interface.
+
+### ip vs. ifconfig
+
+#### `Display ethernet interface 1`
```
Comment (mterwill) [inline _commands/tools/ip.md]: Not code

File: _commands/tools/ip.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+~~~ bash
+$ ifconfig eth1
+
+$ ip addr show eth1
+~~~
+
+####  `Turn eth1 off`
```
Comment (mterwill) [inline _commands/tools/ip.md]: Not code

File: _commands/tools/ip.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+~~~ bash
+$ ifconfig eth1 down
+
+$ ip link set eth1 down
+~~~
+
+####  `Set eth1 IP to 192.168.1.1`
```
Comment (mterwill) [inline _commands/tools/ip.md]: Not code

**Comment by zhecht** [comment]
I'm a little confused here. I'm over 10 commits behind c4cs:master, and 3 commits ahead on my own fork.

The travis is failing because I don't have c4cs-wk6-solutions. Obviously, I forked it before it was uploaded. Do I have to pull the new updates from c4cs:master? How can I do that?

**Comment by mterwill** [comment]
You didn't do anything wrong. Travis fails periodically as documents are released before they're uploaded, resulting in dead links. Fixed the issue + re-ran your build.

**Pull request merged**